tdms/include/hdf5_io/hdf5_base.h

```diff
@@ -7,14 +7,42 @@
 #pragma once
 
 #include <memory>
+#include <stdexcept>
 #include <string>
 #include <vector>
 
 #include <H5Cpp.h>
 
-#include "cell_coordinate.h"
 #include "hdf5_io/hdf5_dimension.h"
 
+/**
+ * @brief Convert from C++ datatype T to a H5::PredType for HDF5 write out.
+ *
+ * TODO: Doesn't support complex doubles. Might need a compount datatype.
+ * Investigate how to read complex doubles first.
```

Comment:
We should probably throw an `std::logic_error("Not implemented")` if we try with `T` as a complex double.

Also `compount` → `compound` (might me my typo).